ai_eval/coach.py

```diff
@@ -0,0 +1,817 @@
+    Your task is to produce a well-structured markdown report in the following format:
+
+    # Evaluation Report
+
+    {% for criterion in scenario_data.evaluation_criteria %}
+        ## {{ criterion.name }}
+        ### Score: (0-5)/5
+        **Rationale**: Provide a rationale for the score, using specific direct quotes from the conversation as evidence.
+    {% endfor %}
+
+    Your response must adhere to this exact structure, and each score must have a detailed rationale that includes at least one direct quote from the chat history.
+    If you cannot find a direct quote, mention this explicitly and provide an explanation.
+""").strip()  # noqa
+
+
+DEFAULT_CONVERSATION_FORMAT = textwrap.dedent("""
+    <conversation>
+        {% for message in messages %}
+            <message>
+                <agent>{{ message.character.name }}</agent>
+                <role>{{ message.character.role }}</role>
+                <content>{{ message.content | escape }}</content>
+            </message>
+        {% endfor %}
+    </conversation>
+""")
+
+
+class CoachAIEvalXBlock(AIEvalXBlock):
+    """
+
+    AI-powered XBlock for simulated conversations with
+    two simulated characters.
+
+    """
+
+    _jinja_env = SandboxedEnvironment(undefined=jinja2.StrictUndefined)
+
+    display_name = String(
+        display_name=_("Display Name"),
+        help=_("Name of the component in the studio"),
+        default="Coached AI Evaluation",
+        scope=Scope.settings,
+    )
+
+    evaluator_prompt = String(
+        display_name=_("Evaluator prompt"),
+        help=_(
+            "Prompt used to instruct the model how to evaluate the learner"
+        ),
+        multiline_editor=True,
+        default=DEFAULT_EVALUATOR_PROMPT,
+        scope=Scope.settings,
+    )
+
+    scenario_title = String(
+        display_name=_("Scenario title"),
+        default="",
+        scope=Scope.settings,
+    )
+
+    initial_message = String(
+        display_name=_("Initial message"),
+        default="",
+        scope=Scope.settings,
+    )
+
+    scenario_data = Dict(
+        display_name=_("Scenario data"),
+        help=_("Arbitrary data accessible in prompt templates"),
+        default={
+            "case_details": "",
+            "learning_objectives": [],
+            "evaluation_criteria": [],
+        },
+        scope=Scope.settings,
+    )
+
+    workspace_title = String(
+        display_name=_("Workspace title"),
+        default=_("Add your answer"),
+        scope=Scope.settings,
+    )
+
+    coach_title = String(
+        display_name=_("Coach title"),
+        default=_("Tutor"),
+        scope=Scope.settings,
+    )
+
+    intro_text = String(
+        display_name=_("Introductory text"),
+        help=_("Optional introductory paragraph shown above the chat panes."),
```

Comment:
This needs more docs: it's interpreted as html on the UI but this is not mentioned. Also, do we want instructors to have to type html?